wallets/index.js

```diff
@@ -1,452 +1,237 @@
-          newClientConfig = Object.assign(newClientConfig, transformedConfig)
-        }
-        // these are stored on the server
-        delete newClientConfig.autoWithdrawMaxFeePercent
-        delete newClientConfig.autoWithdrawThreshold
-        delete newClientConfig.autoWithdrawMaxFeeTotal
-      } catch {
-        valid = false
-      }
-
-      if (valid) {
-        if (priorityOnly) {
-          setClientConfig(newClientConfig)
-        } else {
-          try {
-            // XXX: testSendPayment can return a new config (e.g. lnc)
-            const newerConfig = await wallet.testSendPayment?.(newConfig, { me, logger })
-            if (newerConfig) {
-              newClientConfig = Object.assign(newClientConfig, newerConfig)
+  }, [me?.privates?.walletsUpdatedAt, me?.privates?.vaultKeyHash, refetch])
+
+  useEffect(() => {
+    const loadWallets = async () => {
+      if (!data?.wallets) return
+      // form wallets into a list of { config, def }
+      const wallets = []
+      for (const w of data.wallets) {
+        const def = getWalletByType(w.type)
+        const { vaultEntries, ...config } = w
+        if (isActive) {
+          for (const { key, iv, value } of vaultEntries) {
+            try {
+              config[key] = await decrypt({ iv, value })
+            } catch (e) {
+              console.error('error decrypting vault entry', e)
             }
-          } catch (err) {
-            logger.error(err.message)
-            throw err
           }
-
-          setClientConfig(newClientConfig)
-          logger.ok(wallet.isConfigured ? 'payment details updated' : 'wallet attached for payments')
-          if (newConfig.enabled) wallet.enablePayments()
-          else wallet.disablePayments()
         }
-      }
-    }
 
-    if (hasServerConfig) {
-      let newServerConfig = extractServerConfig(wallet.fields, newConfig)
+        // the specific wallet config on the server is stored in wallet.wallet
+        // on the client, it's stored unnested
+        wallets.push({ config: { ...config, ...w.wallet }, def })
+      }
 
-      let valid = true
-      try {
-        const transformedConfig = await walletValidate(wallet, newServerConfig)
-        if (transformedConfig) {
-          newServerConfig = Object.assign(newServerConfig, transformedConfig)
+      setServerWallets(wallets)
+    }
+    loadWallets()
+  }, [data?.wallets, decrypt, isActive])
+
+  // merge wallets on name like: { ...unconfigured, ...localConfig, ...serverConfig }
+  const wallets = useMemo(() => {
+    const merged = {}
+    for (const wallet of [...walletDefsOnly, ...localWallets, ...serverWallets]) {
+      merged[wallet.def.name] = {
+        def: wallet.def,
+        config: {
+          ...merged[wallet.def.name]?.config,
+          ...Object.fromEntries(
+            Object.entries(wallet.config ?? {}).map(([key, value]) => [
+              key,
+              value ?? merged[wallet.def.name]?.config?.[key]
```

Comment:
If there's a `value` for `key` in a lower priority wallet, but the higher priority config (e.g. server) couldn't be decrypted, we keep `value` for `key`.